internal/cmd/integration-tests/utils.go

```diff
@@ -88,14 +84,17 @@ func runSingleTest(testDir string, port int) {
 		}
 	}
 
+	// sleep for a few seconds before deleting the files to make sure that they are not use anymore
+	time.Sleep(5 * time.Second)
+
```

Comment:
Instead of sleeps, can we have a retry and backoff (without panic on first failure) to remove all files?
Sleeps are notoriously fragile and should be avoided. A slow machine can sometimes hang for 5 seconds.